xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h

```diff
@@ -1266,6 +1266,44 @@ namespace vfs
     //==========================================================================
     ///
     /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve a http response header
+    ///
+    /// @param[in] name         The name of header to retrieve the value for
+    /// @return                 value of requested header, empty string if header name is not found
+    ///
+    std::string CURLGetHeader(const std::string &name)
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before!");
+        return false;
+      }
+      return ::kodi::addon::CAddonBase::m_interface->toKodi->kodi_filesystem->curl_get_header(::kodi::addon::CAddonBase::m_interface->toKodi->kodiBase, m_file, name);
+    }
+    //--------------------------------------------------------------------------
+
+    //==========================================================================
+    ///
+    /// @ingroup cpp_kodi_vfs_CFile
+    /// @brief retrieve a http return code
+    ///
+    /// @return      http return code (e.g. 200)
+    ///
+    int CURLGetReturnCode()
+    {
+      if (!m_file)
+      {
+        kodi::Log(ADDON_LOG_ERROR, "kodi::vfs::CURLCreate(...) needed to call before!");
+        return false;
```

Comment:
return -1; or whatever you want to return in error case. And please don't forget to document that special  value.